### DISCUSSION TITLE
The model name `Subscription` is invalid. It is a reserved name

### DISCUSSION THREAD

**Discussion by alan345** [prisma/prisma#2909] [Q&A]
after running `npx prisma generate`
I have this error: 
```
Error: Schema parsing
error: Error validating model "Subscription": The model name `Subscription` is invalid. It is a reserved name. Please change it.
  -->  schema.prisma:1040
```

Indeed, I have an model called `Subscription` with ~4000 rows. (created with `prisma1`)

Is there a way to bypass this rule? Or should I rename it and change my codebase with the new name?

also adding `  @@map("Subscription_NEW")` is not working

**Comment by janpio** [comment]
> Is there a way to bypass this rule? Or should I rename it and change my codebase with the new name?
> 
> also adding ` @@map("Subscription_NEW")` is not working

`@@map` is the real name in your database, so `@@map("Subscription")`. And then you can call the model something else, e.g. `_Subscription` or similar.

ALthough I am not sure why `Subscription` is still on the denylist.

**Reply by alan345** [reply]
If I change to from `Subscription` to `_Subscription` in `schema.prisma` , I have to change it also in the database?

> ALthough I am not sure why Subscription is still on the denylist.

Do you know when it will be removed from the denylist?

**Reply by janpio** [reply]
No, the `@@map` tells Prisma what the database table is called. It just allows you to use a different name in Prisma Client, that does not clash with the denylist.

What version of Prisma are you using?

**Comment by TomasMoskala** [comment]
Hi I am also using it but lowercase, no issue
schema.prisma
model subscription {
  id_currency                Int
  ...
}

**Reply by ryands17** [reply]
@TomasMoskala did the above way that @janpio mentioned work for you?

**Reply by TomasMoskala** [reply]
I did not run into that trouble at all and generally I try to avoid any complexity.
